v-next/hardhat-ethers/src/internal/signers/signers.ts

```diff
@@ -0,0 +1,280 @@
+import type { HardhatEthersProvider } from "../hardhat-ethers-provider/hardhat-ethers-provider.js";
+import type { NetworkConfig } from "@ignored/hardhat-vnext/types/config";
+import type { BlockTag, TransactionRequest, ethers } from "ethers";
+
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import {
+  assertArgument,
+  getAddress,
+  hexlify,
+  resolveAddress,
+  toUtf8Bytes,
+  TypedDataEncoder,
+} from "ethers";
+
+import { getRpcTransaction } from "../ethers-utils/ethers-utils.js";
+
+import { deepCopy } from "./deep-copy.js";
+import { populate } from "./populate.js";
+
+export class HardhatEthersSigner implements ethers.Signer {
+  readonly #gasLimit: bigint | undefined;
+
+  public readonly address: string;
+  public readonly provider: ethers.JsonRpcProvider | HardhatEthersProvider;
+  public static networkName: string;
+  public static networkConfig: NetworkConfig;
+
+  public static async create(
+    provider: HardhatEthersProvider,
+    networkName: string,
+    networkConfig: NetworkConfig,
+    address: string,
+  ): Promise<HardhatEthersSigner> {
+    this.networkName = networkName;
+    this.networkConfig = networkConfig;
+
+    // depending on the config, we set a fixed gas limit for all transactions
+    let gasLimit: bigint | undefined;
+
+    if (networkName === "hardhat") {
```

Comment:
This changed. Now this check would be `netorkConfig.type !== "http"`.

I think we should create an issue to address this entire section in a follow up pr.

Reply:
I created an issue to track it and I added it in the PR description too.

https://github.com/NomicFoundation/hardhat/issues/5796